foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/utils/BeanUtils.java

```diff
@@ -110,10 +116,36 @@ public static void setContext(ApplicationContext applicationContext) {
   }
 
   public static Class<?> getImplClassFromBean(Object bean) {
```

Comment:
Why not call AopProxyUtils.ultimateTargetClass directly?

Reply:
> Why not call AopProxyUtils.ultimateTargetClass directly?

I am not sure whether should we deal with the 'Advised' case later, so the 'getSingletonTarget' method is copied from AopProxyUtils and put here for later processing.
Do you think using AopProxyUtils#ultimateTargetClass directly is better? I can change this PR if needed.

Reply:
I think use directly is better, if we got specific reasons not use it.